lib/rules/no-invalid-attribute-name.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @author Doug Wade <douglas.b.wade@gmail.com>
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+const xnv = require('xml-name-validator')
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'require valid attribute names',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-invalid-attribute-name.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unexpected: 'Attribute name {{name}} is not valid.'
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    /**
+     * @param {string | VIdentifier} key
+     * @return {string}
+     */
+    const getName = (key) => (typeof key === 'string' ? key : key.name)
+
+    return utils.defineTemplateBodyVisitor(context, {
+      /** @param {VDirective | VAttribute} node */
+      'VAttribute[directive=false]'(node) {
```

Comment:
I think we should check the `v-bind` directive, i.e.

* `v-bind:href="…"` (valid)
* `:href="…"` (valid)
* `v-0abc:href="…"` (invalid)
* `:0abc="…"` (invalid)
* don't check `v-bind="…"`

Please also add tests for all these cases.

Reply:
Thanks so much for including the test cases!

Reply:
Ah, you're right, it seems that I double-clicked the wrong word :sweat_smile: